src/replication.c

```diff
@@ -344,64 +349,80 @@ update_replica_list(int epfd, spec_t *spec, int replica_count) {
 	return replica_count;
 }
 
+void update_volstate(spec_t *spec) {
+	if(((spec->healthy_rcount + spec->degraded_rcount >= spec->consistency_factor) &&
+		(spec->healthy_rcount >= 1))||
+		(spec->healthy_rcount  + spec->degraded_rcount 
+			>= MAX(spec->replication_factor - spec->consistency_factor + 1, spec->consistency_factor))) {
+		spec->ready = true;
+		pthread_cond_broadcast(&spec->rq_cond);
+	} else {
+		spec->ready = false;
+	}
+}
+
+void clear_replica_cmd(spec_t *spec, replica_t *replica, rcmd_t *rep_cmd) {
+	int i;
+	rcommon_cmd_t *rcommq_ptr = rep_cmd->rcommq_ptr;
+	int luworker_id = rcommq_ptr->luworker_id;
+	rcommq_ptr->bitset &= ~(1 << replica->id);
+	//TODO Add check for acks_received in read also, same as write
+	if(rep_cmd->opcode == ZVOL_OPCODE_READ) {
+		rcommq_ptr->status = -1;
+		rcommq_ptr->state = CMD_EXECUTION_COMPLETED;
+		signal_luworker();
+	} else if(rep_cmd->opcode == ZVOL_OPCODE_WRITE) {
+		if(rcommq_ptr->acks_recvd + rcommq_ptr->ios_aborted
+				== rcommq_ptr->copies_sent - 1) {
+			if (rcommq_ptr->acks_recvd < spec->consistency_factor) {
```

Comment:
for sake of consistency, increase ios_aborted here also

Reply:
Lets put a log here that data is committed to few repilcas, but, less than consistency factor, and errored out to client.

Reply:
in this case also, do we need to free iov_base, and remove from pendingq/waitq..?